ibl_pipeline/plotting/behavior_internal.py

```diff
@@ -0,0 +1,389 @@
+                        line=dict(
+                            width=1,
+                            color='orange',
+                            dash='dashdot'
+                        ),
+                        name='85% reference weight',
+                        yaxis='y2',
+                        showlegend=show_res_legend,
+                        legendgroup='weight_ref',
+                        hoverinfo='y'
+                    )
+                )
+
+                data.append(
+                    go.Scatter(
+                        x=[res_start, res_end],
+                        y=[water_res['reference_weight']*0.75,
+                           water_res['reference_weight']*0.75],
+                        mode="lines",
+                        line=dict(
+                            width=1,
+                            color='red',
+                            dash='dashdot'
+                        ),
+                        name='75% reference weight',
+                        yaxis='y2',
+                        showlegend=show_res_legend,
+                        legendgroup='weight_ref',
+                        hoverinfo='y'
+                    )
+                )
+
+            layout = go.Layout(
+                yaxis=dict(
+                    title='Water intake (mL)',
+                    range=yrange_water
+                ),
+                yaxis2=dict(
+                    title='Weight (g)',
+                    overlaying='y',
+                    side='right',
+                ),
+                width=1000,
+                height=500,
+                title=dict(
+                    text='Water intake and weight',
+                    x=0.3,
+                    y=0.9
+                ),
+                xaxis=dict(
+                    title='Date',
+                    range=[d['first_date_str'], d['last_date_str']]
+                ),
+                legend=dict(
+                    x=1.1,
+                    y=0.9,
+                    orientation='v'),
+                barmode='stack',
+                template=dict(
+                    layout=dict(
+                        plot_bgcolor="white"
+                    )
+                )
+            )
+            fig = go.Figure(data=data, layout=layout)
+            water_weight_entry['water_weight'] = fig.to_plotly_json()
+            print('inserting water weight ...')
+            self.insert1(water_weight_entry)
+
+
+# Manually decorate WaterWeight table class.
+context = dict(
+    inspect.currentframe().f_locals,
+    master=behavior_shared.CumulativeSummary,
+    self=WaterWeight,
+    CumulativeSummary=behavior_shared.CumulativeSummary)
+
+if '0.12' in dj.__version__:
+    schema.process_table_class(WaterWeight, context=context)
+elif '0.13' in dj.__version__:
+    schema._decorate_table(WaterWeight, context=context)
+else:
+    raise NotImplementedError('Cannot declare WaterWeight table without DataJoint 0.12 or 0.13.')
+
```

Comment:
Cool, very clever use of this mechanism. Perhaps we can brainstorm to make this more seamless (than a manual code block) but that's for later. Glad it works